recording/docker-compose/docker-compose.yml

```diff
@@ -0,0 +1,17 @@
+version: "3"
+
+services:
+
+  nextcloud-talk-recording:
+    build:
+      context: ..
+      dockerfile: ./docker-compose/Dockerfile
+    shm_size: '2gb'
+    restart: on-failure
+    ports:
+      - "127.0.0.1:8000:8000"
```

Comment:
I would remove the port mapping, as it is not something generic but system specific (for example, in my case `docker-compose up` failed because port 8000 was already used in the host).

Reply:
I would argue that some port mapping should be left, even as a comment. As an end-user it seems to be a more complex task to figure out _why I can't reach out the recording server?_, than _why `docker compose` is yelling about an already used port?_.

Same situation as with DBMS (PostgreSQL, MariaDB, etc.), web servers (lighttpd, NGINX, etc.), and the like. Generally they have a default port mapping already configured, rather than not having it, even if it means colliding with an already-running service.

Reply:
Mmm, but, if I am not mistaken, other containers should be able to access the recording server using the Docker network automatically created by Docker compose, it should not be necessary to expose the recording server directly on the host. And if it needs to be exposed directly in the host because the other services are not managed with Docker I think that would be specific to that environment, not something generic that should be included by default in the Docker compose file for everyone.

Having said all that the problem could be that I am not too familiar with how Docker compose files are written :-)

@SystemKeeper Do you have any opinion on all this? Thanks! :-)

Reply:
I think it heavily depends on how you're using docker / docker-compose. When you have all containers in one compose file, a default network is created allowing the containers to reach each other. This is not the case here (obviously) so we either need to create a network, so other containers can join that network, or expose the recording server externally. Therefore I would agree that both does not really make sense, especially since the example is exposing only to the local host. 

Since this docker-compose file most likely will be used as a blueprint by the users (and not "as is"), I would suggest that we keep the network and add a note about the reachability so that beginners have a starting point. Something like this:

```yaml
version: "3.9"

services:

  nextcloud-talk-recording:
    build:
      context: ..
      dockerfile: ./docker-compose/Dockerfile
    init: true
    shm_size: '2gb'
    restart: on-failure
    # By default the recording server is reachable through the network "nextcloud-talk-recording"
    # Depending on your setup (if you need to reach the recording server externally for example) you might need
    # to expose the used ports to the host machine, e.g.:
    #ports:
    #  - "8000:8000"
    networks:
      - nextcloud-talk-recording

networks:
  nextcloud-talk-recording:
``` 

What do you both think?
After that I would say fine to merge from my side 😄

Reply:
> [...]
> What do you both think?

I agree :-) Thanks for the suggestion!

Reply:
@mishamosher what do you think? Wanna do the adjustments or shall we take over to get this merged? :)

Reply:
@SystemKeeper @danxuliu 

I'm in a bad place with time availability right now. If this can wait till the second week of August, I can make the changes.

If one of you wants to make the changes and merge them, I'm all for it.